sprs/src/num_kinds.rs

```diff
@@ -2,13 +2,106 @@
 //! or a complex.
 
 use num_complex::{Complex32, Complex64};
-use std::fmt;
+use num_traits::{Num, NumCast, One, ToPrimitive, Zero};
+use std::{
+    fmt,
+    ops::{Add, Div, Mul, Rem, Sub},
+};
+/// the type for Pattern data, it's special which contains no data
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct Pattern;
+
+impl Add for Pattern {
+    type Output = Pattern;
+    fn add(self, _other: Pattern) -> Pattern {
+        Pattern {}
+    }
+}
+impl Zero for Pattern {
+    fn zero() -> Self {
+        Pattern {}
+    }
+    fn is_zero(&self) -> bool {
+        true
```

Comment:
This should be `false` as absence marks a zero

Reply:
I have deleted this trait.